src/server/node_services/nodes_monitor.js

```diff
@@ -2657,47 +2515,7 @@ class NodesMonitor extends EventEmitter {
         }
 
         const storage_mode = host_item.storage_nodes_mode;
-
-        const s3_node_modes = [
-            'DELETING',
-            'OFFLINE',
-            'UNTRUSTED',
-            'INITIALIZING',
-            'DECOMMISSIONED',
-            'OPTIMAL',
-            'HTTP_PORT_ACCESS_ERROR',
-            'HTTP_SRV_ERRORS',
-        ];
-        s3_nodes.forEach(node => {
-            if (s3_node_modes.indexOf(node.mode) === -1) {
-                node.mode = 'HTTP_SRV_ERRORS';
-            }
-        });
-
-        const s3_mode = s3_nodes.length ? s3_nodes[0].mode : 'DECOMMISSIONED';
-        host_item.s3_nodes_mode = s3_mode;
-        const storage_priority = mode_priority[storage_mode];
-        const s3_priority = mode_priority[s3_mode];
-
-        if (s3_mode === 'DECOMMISSIONED') {
-            host_item.mode = storage_mode;
-        } else if (storage_mode === 'DECOMMISSIONED') {
-            host_item.mode = s3_mode;
-        } else if (s3_mode === 'DELETING' || storage_mode === 'DELETING') {
-            host_item.mode = 'DELETING';
-        } else if (s3_mode === storage_mode) {
-            host_item.mode = s3_mode;
-        } else if (s3_priority > storage_priority) {
-            host_item.mode = s3_mode === 'OFFLINE' ? 'S3_OFFLINE' : s3_mode;
-        } else if (storage_priority > s3_priority) {
-            host_item.mode = storage_mode === 'OFFLINE' ? 'STORAGE_OFFLINE' : storage_mode;
-        } else {
-            host_item.mode = mode_by_priority[s3_priority];
-            // in case the mode is optimal, check if there are blocked ports
-            if (host_item.mode === 'OPTIMAL' && storage_mode === 'N2N_PORTS_BLOCKED') {
-                host_item.mode = 'N2N_PORTS_BLOCKED';
-            }
-        }
+        host_item.mode = storage_mode;
```

Comment:
why did you remove mode_priority ?

Reply:
After the change, there is no conflict between storage_mode and endpoint_mode because there is no endpoint mode. so the priority of each is not relevant anymore.

Reply:
It was also between different drives with different modes.
But for, for now, we don't foresee a multi-drive so ok.

Reply:
No, it was not used for that. The only use was between the aggr storage nodes mode and the aggr endpoint nodes mode. we still support multi-drive host with a proper mode